### PR TITLE
Support for deployment of certain contracts in specific addresses

### PR DESCRIPTION
**Pull request by ggrieco-tob**
This PR adds support for deployment of certain contracts in specific address. A simple example:

```solidity
contract A {
  uint256 public x = 128;
}

contract Test {
  function test(uint256 y) public {
    assert(A(address(0x42)).x() == y);
  }
}
```

with the following config file:

```yaml
deploy: [["0x42", "A"]]
```

To test:

```bash
$ echidna-test deploy.sol --test-mode assertion --config deploy.yaml --contract Test
```

### CODE REVIEW

**Comment by hellwolf** [comment]
thank you! do you recommend me to build echidna-test from this branch? Is it cabal based?

**Comment by ggrieco-tob** [comment]
It is using stack (which will somehow use cabal). You can grab the binaries directly from our CI if you want to quickly test it.

**Comment by ggrieco-tob** [comment]
@patrickd- could this PR help you to simplify your use of Echidna?

**Comment by patrickd-** [comment]
hmm, no, haven't had the need for this one yethmm, no, haven't had the need for this one yet

**Comment by hellwolf** [comment]
tried the binary, it works!!

Branch: https://github.com/superfluid-finance/protocol-monorepo/pull/708

How to run (I downloaded the CI build to ~/tmp):
```
$ ECHIDNA=~/tmp/echidna-test ./fuzz.sh CFAFuzzer
```

What I did, in config:

```
deploy: [["0x1820a4B7618BdE71Dce8cdc73aAB6C95905faD24", "ERC1820RegistryMock"]]
```

Caveat though:

Usually the ERC1820Registry is distributed as an exact binary, e.g.:  https://github.com/superfluid-finance/protocol-monorepo/blob/dev/packages/ethereum-contracts/scripts/artifacts/ERC1820Registry.json But I guess the syntax proposed here wouldn't be able to use a raw binary like this.

Great stuff anyhow! Thank you!

**Comment by ggrieco-tob** [comment]
We can definitely allow to use a precompiled contract, instead of a contract name. It will also be useful for differential fuzzing

File: lib/Echidna/Solidity.hs [hunk 1]
```diff
@@ -223,7 +223,7 @@ loadSpecified name cs = do
       -- main contract deployment
       let deployment = execTx $ createTxWithValue bc d ca (fromInteger unlimitedGasPerBlock) (w256 $ fromInteger balc) (0, 0)
       vm3 <- execStateT deployment vm2
-      when (isNothing $ currentContract vm2) (throwM $ DeploymentFailed ca)
+      when (isNothing $ currentContract vm3) (throwM $ DeploymentFailed ca)
```
Comment (hellwolf) [inline lib/Echidna/Solidity.hs:226]: aha, we need linear type!! :D

**Comment by hellwolf** [comment]
Happy with this so far, this has been integrated in our fuzzing testsuite.

Any plan to merge this soon and release?

**Comment by ggrieco-tob** [comment]
We need to add some tests here, but if you don't have more suggestions, it can be merged as soon as next week

**Pull request merged**